File: Data Structures & Algorithms/majority-element-ii/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    majorityElement(nums) {

        const n = nums.length;

        const res = new Set();

        outer: for(let i = 0; i < n; i += 1) {

            // The items was previously identified as a majority element
            if(res.has(nums[i])) {
                continue;
            }
            
            let count = 0;
            for(let j = 0; j < n; j += 1) {

                if(nums[i] === nums[j]) {
                    count += 1;
                    if(count > n / 3) {
                        res.add(nums[i]);
                        if(res.size === 2) {
                            // There can be no more than 2 majority elements
                            break outer;
                        } else {
                            // break from inner
                            break;
                        }   
                    }
                }

            }

        }

        return Array.from(res);

    }
}

File: Data Structures & Algorithms/majority-element/submission-6.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:

        if len(nums) == 0:
            return -1

        maj_el = nums[0]
        count = 1

        for i in range(1, len(nums)):
            curr_el = nums[i]

            if curr_el != maj_el:
                count -= 1
                if count < 0:
                    maj_el = curr_el
                    count = 1
            else:
                count += 1

        return maj_el